cmake/Modules/FindJack.cmake

```diff
@@ -1,82 +1,78 @@
-# - Try to find jack-2.6
-# Once done this will define
+# * Try to find jack-2.6 Once done this will define
 #
-#  JACK_FOUND - system has jack
-#  JACK_INCLUDE_DIRS - the jack include directory
-#  JACK_LIBRARIES - Link these to use jack
-#  JACK_DEFINITIONS - Compiler switches required for using jack
+# JACK_FOUND - system has jack JACK_INCLUDE_DIRS - the jack include directory
+# JACK_LIBRARIES - Link these to use jack JACK_DEFINITIONS - Compiler switches
+# required for using jack
 #
-#  Copyright (c) 2008 Andreas Schneider <mail@cynapses.org>
-#  Modified for other libraries by Lasse Kärkkäinen <tronic>
+# Copyright (c) 2008 Andreas Schneider <mail@cynapses.org> Modified for other
+# libraries by Lasse Kärkkäinen <tronic>
 #
-#  Redistribution and use is allowed according to the terms of the New
-#  BSD license.
-#  For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+# Redistribution and use is allowed according to the terms of the New BSD
+# license. For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 #
 
-if (JACK_LIBRARIES AND JACK_INCLUDE_DIRS)
+if(JACK_LIBRARIES AND JACK_INCLUDE_DIRS)
   # in cache already
   set(JACK_FOUND TRUE)
-else (JACK_LIBRARIES AND JACK_INCLUDE_DIRS)
-  # use pkg-config to get the directories and then use these values
-  # in the FIND_PATH() and FIND_LIBRARY() calls
-  if (${CMAKE_MAJOR_VERSION} EQUAL 2 AND ${CMAKE_MINOR_VERSION} EQUAL 4)
+else(JACK_LIBRARIES AND JACK_INCLUDE_DIRS)
+  # use pkg-config to get the directories and then use these values in the
+  # FIND_PATH() and FIND_LIBRARY() calls
+  if(${CMAKE_MAJOR_VERSION} EQUAL 2 AND ${CMAKE_MINOR_VERSION} EQUAL 4)
     include(UsePkgConfig)
     pkgconfig(jack _JACK_INCLUDEDIR _JACK_LIBDIR _JACK_LDFLAGS _JACK_CFLAGS)
-  else (${CMAKE_MAJOR_VERSION} EQUAL 2 AND ${CMAKE_MINOR_VERSION} EQUAL 4)
+  else(${CMAKE_MAJOR_VERSION} EQUAL 2 AND ${CMAKE_MINOR_VERSION} EQUAL 4)
     find_package(PkgConfig)
-    if (PKG_CONFIG_FOUND)
+    if(PKG_CONFIG_FOUND)
       pkg_check_modules(_JACK jack)
```

Comment:
Linking JackWeakAPI for macOS and Windows requires linking all of the libraries specified by pkgconfig, namely `dl` in addition to `jack`. So that would be using `_JACK_LINK_LIBRARIES` from this `pkg_check_modules` in preference to doing `find_library` for the single `jack` library. Take a look at https://github.com/PortAudio/portaudio/blob/master/cmake/modules/FindJACK.cmake for example.

Reply:
Come to think of it, you may want to copy those FindJACK.cmake and FindRegex.cmake modules directly from PortAudio.